File: python/sender.py
#!/usr/bin/env python2
'''This is a simple message sender application that uses the rethinkmq
library.

It receives a message at the command line and sends it to the
specified topic
'''

import sys
import logging

from rethinkmq import MQConnection

USAGE = '''\
Usage: {0} [-v] <topic> <message>

Examples:

  Broadcast the temperature in Mountain View
    {0} weather.us.ca.mountainview 74
  Broadcast the temperature in Mountain View
    {0} weather.uk.conditions "A bit rainy"

'''

def main(argv):
    if len(argv) < 3:
        print USAGE.format(argv[0])
        return

    if argv[1] == '-v':
        logging.root.setLevel(logging.DEBUG)
        logging.basicConfig()
        argv = argv[1:]

    # Publish on the topic (will create the table if necessary)
    MQConnection(db='MQ').exchange('messages').topic(argv[1]).publish(argv[2])


if __name__ == '__main__':
    main(sys.argv)

File: ruby/sender.rb
#!/usr/bin/env ruby
# This is a simple message sender application that uses the rethinkmq
# library.

# It receives a message at the command line and sends it to the
# specified topic

require 'logger'
require_relative 'rethinkmq'

$USAGE = <<USAGE
Usage: #{$PROGRAM_NAME} [-v] <topic> <message>

Examples:

  Broadcast the temperature in Mountain View
    #{$PROGRAM_NAME} weather.us.ca.mountainview 74
  Broadcast the temperature in Mountain View
    #{$PROGRAM_NAME} weather.uk.conditions "A bit rainy"

USAGE

def main
  if ARGV[0] == '-h'
    puts $USAGE
    return
  elsif ARGV[0] == '-v'
    ARGV.shift
  else
    $RMQLogger.level = Logger::ERROR
  end

  MQConnection.new('MQ').exchange('messages').topic(ARGV[0]).publish(ARGV[1])
end

main
